Utilities/Dox/PythonScripts/FileManGlobalDataParser.py

```diff
@@ -790,59 +790,71 @@ def run(args):
 
-  glbDataParser.patchDir = args.patchRepositDir
-  htmlGen = FileManDataToHtml(crossRef, glbDataParser.schemaParser, args.outDir, _doxURL, _vivianURL)
-  isolatedFiles = glbDataParser.schemaParser.isolatedFiles
-  if not args.all or set(args.fileNos).issubset(isolatedFiles):
+  if not args.all:
+    isolatedFiles = glbDataParser.schemaParser.isolatedFiles
+    assert set(args.fileNos).issubset(isolatedFiles)
+    assert set(args.fileNos).issubset(glbDataParser.allFiles)
     for fileNo in args.fileNos:
-      for gdFile in glbDataParser.allFiles[fileNo]['path']:
-        logger.info("Parsing file: %s at %s" % (fileNo, gdFile))
-        glbDataParser.parseZWRGlobalFileBySchemaV2(gdFile, fileNo)
-      htmlGen.outputFileManDataAsHtml(fileNo, glbDataParser)
-      del glbDataParser.outFileManData[fileNo]
-      gc.collect()
+        assert fileNo in glbDataParser.globalLocationMap
+
+    processFiles(args.fileNos)
   else:
-    # Generate all required files
-    sccSet = glbDataParser.schemaParser.sccSet
-    fileSet = set(args.fileNos)
-    for idx, value in enumerate(sccSet):
-      fileSet.difference_update(value)
-      if not fileSet:
-        break
-    for i in xrange(0,idx+1):
-      fileSet = fileSet.union(sccSet[i])
-      fileSet &= set(glbDataParser.allFiles.keys())
-    if len(fileSet) > 1:
-      for file in fileSet:
-        for zwrFile in glbDataParser.allFiles[file]['path']:
-          globalSub = glbDataParser.allFiles[file]['name']
-          glbDataParser.generateFileIndex(zwrFile, file)
+    # glbDataParser.schemaParser.sccSet is a list of sets
+    # Combine into one set
+    fileSet = set()
+    for i in glbDataParser.schemaParser.sccSet:
+      fileSet = fileSet.union(i)
+
+    fileSet.add('101')    #Protocol
+    fileSet.add('8994')   #Remote Procedure
+    fileSet.add('19')     #Option
+    fileSet.add('779.2')  #HLO Application
+    fileSet.add('9.7')    #Install
+    fileSet.add('.5')     #Function
+    fileSet.add('409.61') #List Template
+    fileSet.add('19.1')   #Security Key
+    fileSet.add('9.2')    #Help Frame
+    fileSet.add('.403')   #Form
+    fileSet.add('.401')   #Sort Template
+    fileSet.add('771')    #HL7 APPLICATION PARAMETER
+
+    # Only include files that are in fileset and glbDataParser.allFiles.keys()
+    fileSet &= set(glbDataParser.allFiles.keys())
+
     for file in fileSet:
-        for zwrFile in glbDataParser.allFiles[file]['path']:
-          logger.info("Parsing file: %s at %s" % (file, zwrFile))
-          globalSub = glbDataParser.allFiles[file]['name']
-          glbDataParser.parseZWRGlobalFileBySchemaV2(zwrFile, file)
-        htmlGen.outputFileManDataAsHtml(file, glbDataParser)
-        glbDataParser.outFileManData.pop(file)
-        gc.collect()
+      for zwrFile in glbDataParser.allFiles[file]['path']:
+        glbDataParser.generateFileIndex(zwrFile, file)
+    processFiles(fileSet)
+
   glbDataParser.outRtnReferenceDict()
 
+def processFiles(files):
```

Comment:
A test run complained here that glbDataParser wasn't defined in this function at line 851.  Should probably pass it into the function or change glbDataParser to be a part of "self".

> 142: CMake Error at /opt/VistA-docs/Docs/CallerGraph/FileManGlobalDataParser.cmake:36 (message):
> 142:   Error running FileMan Global Data Parser:
> 142:
> 142:
> 142:
> 142:   Traceback (most recent call last):
> 142:
> 142:     File "/opt/VistA/Utilities/Dox/PythonScripts/FileManGlobalDataParser.py", line 905, in <module>
> 142:       main()
> 142:     File "/opt/VistA/Utilities/Dox/PythonScripts/FileManGlobalDataParser.py", line 902, in main
> 142:       run(result)
> 142:     File "/opt/VistA/Utilities/Dox/PythonScripts/FileManGlobalDataParser.py", line 845, in run
> 142:       processFiles(fileSet)
> 142:     File "/opt/VistA/Utilities/Dox/PythonScripts/FileManGlobalDataParser.py", line 851, in processFiles
> 142:       for zwrFile in glbDataParser.allFiles[file]['path']:
> 142:
> 142:   NameError: global name 'glbDataParser' is not defined
> 142:

Reply:
Fixed